File: samples/python/81.skills-skilldialog/dialog-skill-bot/bots/skill_bot.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

from botbuilder.core import (
    Bot,
    ConversationState,
    TurnContext,
)
from botbuilder.dialogs import Dialog, DialogExtensions


class SkillBot(Bot):
    def __init__(self, conversation_state: ConversationState, main_dialog: Dialog):
        self._conversation_state = conversation_state
        self._main_dialog = main_dialog

    async def on_turn(self, context: TurnContext):
        await DialogExtensions.run_dialog(
            self._main_dialog,
            context,
            self._conversation_state.create_property("DialogState"),
        )

        # Save any state changes that might have occurred during the turn.
        await self._conversation_state.save_changes(context)

File: samples/csharp_dotnetcore/81.skills-skilldialog/DialogSkillBot/Bots/SkillBot.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Threading;
using System.Threading.Tasks;
using Microsoft.Bot.Builder;
using Microsoft.Bot.Builder.Dialogs;

namespace Microsoft.BotBuilderSamples.DialogSkillBot.Bots
{
    public class SkillBot<T> : ActivityHandler
        where T : Dialog
    {
        private readonly ConversationState _conversationState;
        private readonly Dialog _mainDialog;

        public SkillBot(ConversationState conversationState, T mainDialog)
        {
            _conversationState = conversationState;
            _mainDialog = mainDialog;
        }

        public override async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default)
        {
            await _mainDialog.RunAsync(turnContext, _conversationState.CreateProperty<DialogState>("DialogState"), cancellationToken);

            // Save any state changes that might have occurred during the turn.
            await _conversationState.SaveChangesAsync(turnContext, false, cancellationToken);
        }
    }
}
